core/src/main/java/org/apache/accumulo/core/iterators/WrappingIterator.java

```diff
@@ -58,21 +58,21 @@ protected void setSource(SortedKeyValueIterator<Key,Value> source) {
   public Key getTopKey() {
     if (seenSeek == false)
       throw new IllegalStateException("never been seeked");
-    return getSource().getTopKey();
+    return source.getTopKey();
```

Comment:
I experimented with making the VersioningIterator and Filter not call WrappingIterator.getSource() multiple times in their loops.  I saw a noticeable difference, like 5% increase for the VersioningIterator.  I change the code to call getSource() once before the loop and store the source in a local var.